pkg/docker/docker.go

```diff
@@ -439,7 +538,7 @@ func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
 	}
 
 	image, err := d.CheckImage(name)
-	if err != nil && err.(errors.Error).Details != docker.ErrNoSuchImage {
+	if err != nil && dockerapi.IsErrImageNotFound(err) {
```

Comment:
I discovered after running the test suites (and maybe remember) why a ! had been put there (though the code I had there is still not correct) .... best as I can tell, there is a bug with engine-api or docker.  The inspect code (which is the only way we can get the image metadata ... pull does not provide it) will look for a `http.StatusNotFound` on the http get for inspect, and create the engine-api image not found error, however, when we access an image not available locally, we get this for the error message are result from `dockerapi.IsErrImageNotFound`:
```
no such image: "openshift/php-55-centos7:latest" and result from engine-api error check false
```
Looking at the message for the engine-api image not found error, the first n is captialized ... i.e. `No such image`.  So perhaps that is another clue that it is not the error type.

Bottom line, short term, I have to punt on `dockerapi.IsErrImageNotFound` and check the message string instead (yuck).  There is not an error message / type / method in engine-api that has `no such image` (lower case `n`) for the error.